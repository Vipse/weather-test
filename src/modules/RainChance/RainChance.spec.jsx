import React from "react";
import { shallow } from "enzyme";
import { RainChance } from "./index";
import renderer from "react-test-renderer";

const props = {
    loadingForecast: true,
    forecast: [],
    onGetForecast: jest.fn()
};

describe("RainChance component test", () => {
  it("renders without crashes", () => {
      const wrapper = shallow(<RainChance {...props} />);
      expect(wrapper.exists()).toBeTruthy();
      expect(props.onGetForecast).toBeCalledTimes(1);
  });

  it("renders styles correctly", () => {
      const tree = renderer.create(
          <RainChance {...props} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
  });
});
