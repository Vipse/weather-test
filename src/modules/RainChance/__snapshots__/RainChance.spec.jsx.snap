// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RainChance component test renders styles correctly 1`] = `
<div
  className="ant-spin ant-spin-lg ant-spin-spinning forecast-spinner"
>
  <span
    className="ant-spin-dot ant-spin-dot-spin"
  >
    <i
      className="ant-spin-dot-item"
    />
    <i
      className="ant-spin-dot-item"
    />
    <i
      className="ant-spin-dot-item"
    />
    <i
      className="ant-spin-dot-item"
    />
  </span>
</div>
`;
